File: yejin/22_12_01_02.py
# 캐릭터의 좌표
def solution(keyinput, board):
    r = 0
    u = 0
# 가로측 좌표 r과 세로측 좌표 u를 정한다.
    for i in keyinput:
        if i == "left":
            r -= 1
        elif i == "right":
            r += 1
        elif i == "up":
            u += 1
        else:
            u -= 1
# board 값을 통해 좌표의 증가 범위를 제한해준다.
        if r > board[0]//2:
            r = board[0]//2
        elif r < board[0]//2*-1:
            r = board[0]//2*-1
        if u > board[1]//2:
            u = board[1]//2
        elif u < board[1]//2*-1:
            u = board[1]//2*-1  
    return [r, u]

File: seungbeom/22_12_01_02.java
// https://school.programmers.co.kr/learn/courses/30/lessons/120861

class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[2];

        for (String singleInput : keyinput) {
            switch (singleInput) {
                case "up":
                    answer[1] = Math.min(++answer[1], board[1]/2);
                    break;
                case "down":
                    answer[1] = Math.max(--answer[1], -board[1]/2);
                    break;
                case "right":
                    answer[0] = Math.min(++answer[0], board[0]/2);
                    break;
                case "left":
                    answer[0] = Math.max(--answer[0], -board[0]/2);
                    break;
                default:
                    break;
            }
        }

        return answer;
    }
}